src/lib/server/searchWeb.ts

```diff
@@ -0,0 +1,39 @@
+import { SERPAPI_KEY } from "$env/static/private";
+import type { Message } from "$lib/types/Message";
+
+import { getJson } from "serpapi";
+import type { GoogleParameters } from "serpapi";
+import { generateFromDefaultEndpoint } from "./generateFromDefaultEndpoint";
+
+// Show result as JSON
+export async function searchWeb(query: string) {
+	const params = {
+		q: query,
+		hl: "en",
+		gl: "us",
+		google_domain: "google.com",
+		api_key: SERPAPI_KEY,
+	} satisfies GoogleParameters;
+
+	// Show result as JSON
+	const response = await getJson("google", params);
+
+	return response;
+}
+
+export async function getQueryFromPrompt(messages: Pick<Message, "from" | "content">[]) {
+	let prompt =
+		"<|prompter|>The following messages were written by a user, trying to answer a question.\n";
+
+	messages
+		.filter((message) => message.from === "user")
+		.forEach((message) => {
+			prompt += `<|${message.from}|> ${message.content}</s>`;
```

Comment:
```suggestion
			prompt += `<|user|> ${message.content}${model.messageEndToken ?? ""}`;
```

?

(if `message.from` was from the assistant, we'd want to use `defaultModel.assistantMessageToken` instead)

We also probably want to default `model.messageEndToken` to `""` in `server/models.ts` to simplify code here & in `buildPrompt`